embd.go

```diff
@@ -72,20 +78,50 @@ func run() error {
 		Args:  os.Args[1:],
 		Pkg:   *pkg,
 		Files: map[string]File{},
+		Dirs:  map[string]map[string]File{},
 	}
 	for _, path := range flag.Args() {
-		// TODO(akavel): support directories
-		f, err := NewFile(path)
+		path := filepath.ToSlash(path)
+		info, err := os.Stat(path)
 		if err != nil {
 			return err
 		}
-		if old, exists := contents.Files[f.VarName]; exists {
-			return fmt.Errorf(
-				"generated variable name conflict: '%s' resolves to"+
-					"the same variable name %s as '%s'",
-				f.Path, f.VarName, old.Path)
+
+		if info.IsDir() {
+			dir, err := ioutil.ReadDir(path)
+			if err != nil {
+				return err
+			}
+
+			k := "Dir" + normalize(path)
+			if _, exists := contents.Dirs[k]; exists {
+				return fmt.Errorf("directory %s was resolved to variable %s. But last's already found", path, info.Name())
+			}
+
+			files := map[string]File{}
+			for _, info := range dir {
+				f, err := NewFile(path + "/" + info.Name())
+				if err != nil {
+					return err
+				}
+				files[f.VarName] = f
```

Comment:
Add a duplicates check here, similar as in lines 115-120, but on `files` instead of on `contents`.